tests/countries/test_benin.py

```diff
@@ -0,0 +1,230 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.countries.benin import Benin, BJ, BEN
+from tests.common import CommonCountryTests
+
+
+class TestBenin(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(1960, 2050)
+        super().setUpClass(Benin, years=years, years_non_observed=years)
+        cls.no_estimated_holidays = Benin(years=years, islamic_show_estimated=False)
+
+    def test_country_aliases(self):
+        self.assertAliases(Benin, BJ, BEN)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(Benin(years=1959))
+
+    def test_new_years_day(self):
+        name = "Jour de l'An"
+        self.assertHolidayName(
+            name,
+            "2020-01-01",
+            "2021-01-01",
+            "2022-01-01",
+            "2023-01-01",
+            "2024-01-01",
+            "2025-01-01",
+        )
+        self.assertHolidayName(name, range(1960, 2050))
```

Comment:
```suggestion
        self.assertHolidayName("Jour de l'An", (f"{year}-01-01" for year in range(1960, 2050)))
```
For fixed date holidays.